### PR TITLE
[SPARK-33556][ML] Add array_to_vector function for dataframe column

### PR DESCRIPTION
**Pull request by WeichenXu123**
### What changes were proposed in this pull request?

Add array_to_vector function for dataframe column

### Why are the changes needed?
Utility function for array to vector conversion.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
scala unit test & doctest.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/functions.scala [hunk 1]
```diff
@@ -72,6 +72,21 @@ object functions {
     }
   }
 
+  private val arrayToVectorUdf = udf { array: Seq[Double] =>
+    Vectors.dense(array.toArray)
+  }
+
+  /**
+   * Converts a column of array&lt;NumericType&gt type into a column of MLlib
```
Comment (HyukjinKwon) [inline mllib/src/main/scala/org/apache/spark/ml/functions.scala:80]: I would just say as prose:

```
Converts a column of array of a numeric type into a column of dense vectors in MLlib.
```

File: mllib/src/main/scala/org/apache/spark/ml/functions.scala [hunk 2]
```diff
@@ -72,6 +72,21 @@ object functions {
...
+   * dense vectors
+   * @param v: the column of array&lt;NumericType&gt type
+   * @return a column of type "org.apache.spark.ml.linalg.Vector"
```
Comment (HyukjinKwon) [inline mllib/src/main/scala/org/apache/spark/ml/functions.scala:83]: nit: `"org.apache.spark.ml.linalg.Vector"` -> `` `org.apache.spark.ml.linalg.Vector` ``

File: python/pyspark/ml/functions.py [hunk 3]
```diff
@@ -69,6 +69,31 @@ def vector_to_array(col, dtype="float64"):
         sc._jvm.org.apache.spark.ml.functions.vector_to_array(_to_java_column(col), dtype))
 
 
+def array_to_vector(col):
+    """
+    Converts a column of array<NumericType> type into a column of MLlib dense vectors
+
```
Comment (HyukjinKwon) [inline python/pyspark/ml/functions.py:75]: @WeichenXu123, after SPARK-32085, now PySpark uses [NumPy documentation style](https://numpydoc.readthedocs.io/en/latest/format.html).

**Review by HyukjinKwon — COMMENTED** [review]
Shall we add a simple test at `mllib/src/test/scala/org/apache/spark/ml/FunctionsSuite.scala`? Looks fine otheriwse.

**Comment by HyukjinKwon** [comment]
cc @zero323 in case you're interested in it. We might have to add the R version as well with a separate JIRA.

**Comment by zero323** [comment]
> cc @zero323 in case you're interested in it. We might have to add the R version as well with a separate JIRA.

Thanks @HyukjinKwon . I'll take care of that, once this PR is merged.

@WeichenXu123 Could you also add type hints to [functions.pyi](https://github.com/apache/spark/blob/master/python/pyspark/ml/functions.pyi)?

File: python/pyspark/ml/functions.py [hunk 4]
```diff
@@ -69,6 +69,40 @@ def vector_to_array(col, dtype="float64"):
         sc._jvm.org.apache.spark.ml.functions.vector_to_array(_to_java_column(col), dtype))
 
 
+def array_to_vector(col):
+    """
+    Converts a column of array<NumericType> type into a column of MLlib dense vectors
```
Comment (HyukjinKwon) [inline python/pyspark/ml/functions.py:74]:
```suggestion
    Converts a column of array of numeric type into a column of dense vectors in MLlib
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by zero323** [comment]
FYI I opened a [PR](https://github.com/apache/spark/pull/30561) with Spark variant..

**Pull request closed without merging**